larq_zoo/dorefanet.py

```diff
@@ -0,0 +1,183 @@
+import tensorflow as tf
+import larq as lq
+from larq_zoo import utils
+from zookeeper import registry, HParams
+
+
+@registry.register_model
+def dorefa_net(hparams, input_shape, num_classes, input_tensor=None, include_top=True):
+    def conv_block(x, features, kernel_size, strides=1, pool=False, padding="same"):
+        x = lq.layers.QuantConv2D(
+            features,
+            kernel_size=kernel_size,
+            strides=strides,
+            padding="same",
+            input_quantizer=hparams.input_quantizer,
+            kernel_quantizer=hparams.kernel_quantizer,
+            kernel_constraint=None,
+            use_bias=False,
+        )(x)
+        x = tf.keras.layers.BatchNormalization(scale=False, momentum=0.9, epsilon=1e-4)(
+            x
+        )
+        if pool:
+            x = tf.keras.layers.MaxPool2D(pool_size=3, strides=2, padding=padding)(x)
+        return x
+
+    def dense_block(x, units):
+
+        x = lq.layers.QuantDense(
+            units,
+            input_quantizer=hparams.input_quantizer,
+            kernel_quantizer=hparams.kernel_quantizer,
+            kernel_constraint=None,
+            use_bias=False,
+        )(x)
+        x = tf.keras.layers.BatchNormalization(scale=False, momentum=0.9, epsilon=1e-4)(
+            x
+        )
+        return x
+
+    # get input
+    img_input = utils.get_input_layer(input_shape, input_tensor)
+
+    # feature extractor
+    out = tf.keras.layers.Conv2D(
+        96, kernel_size=12, strides=4, padding="valid", use_bias=True
+    )(img_input)
+    out = conv_block(out, features=256, kernel_size=5, pool=True)
+    out = conv_block(out, features=384, kernel_size=3, pool=True)
+    out = conv_block(out, features=384, kernel_size=3)
+    out = conv_block(out, features=256, kernel_size=3, padding="valid", pool=True)
+
+    # classifier
+    if include_top:
+        out = tf.keras.layers.Flatten()(out)
+        out = dense_block(out, units=4096)
+        out = dense_block(out, units=4096)
+        out = tf.keras.layers.Activation("clip_by_value_activation")(out)
+        out = tf.keras.layers.Dense(num_classes, use_bias=True)(out)
+        out = tf.keras.layers.Activation("softmax")(out)
+
+    return tf.keras.Model(inputs=img_input, outputs=out)
+
+
+@lq.utils.register_keras_custom_object
+@lq.utils.set_precision(1)
+def magnitude_aware_sign_unclipped(x):
+    r"""
```

Comment:
No need for raw strings since we don't use LaTeX in the doc string
```suggestion
    """
```